### PR TITLE
Add basic IScriptable methods

### PR DESCRIPTION
**Pull request by Roms1383**
This small PR adds `IScriptable` most basic methods :
- `GetClassName`
- `IsA`
- `IsExactlyA`

> I'm not sure how to treat `DetectScriptableCycles` so I left it aside.

to avoid having to create a newtype pattern like so:
```rs
use red4ext_rs::{
    conv::ClassType,
    macros::redscript_import,
    types::{CName, Ref},
};

#[repr(transparent)]
pub struct IScriptable(red4ext_rs::types::IScriptable);

impl ClassType for IScriptable {
    type BaseClass = red4ext_rs::types::IScriptable;

    const NAME: &'static str = red4ext_rs::types::IScriptable::NAME;
    const NATIVE_NAME: &'static str = red4ext_rs::types::IScriptable::NATIVE_NAME;
}

#[redscript_import]
impl IScriptable {
    /// public native func GetClassName() -> CName
    #[redscript(native)]
    pub fn get_class_name(self: &Ref<Self>) -> CName;
    /// public native func IsA(className: CName) -> Bool
    #[redscript(native)]
    pub fn is_a(self: &Ref<Self>, class_name: CName) -> bool;
    /// public native func IsExactlyA(className: CName) -> Bool
    #[redscript(native)]
    pub fn is_exactly_a(self: &Ref<Self>, class_name: CName) -> bool;
}
```

especially when deriving children classes like so:
```rs
use crate::interop::iscriptable::IScriptable;
use red4ext_rs::types::Ref;
use red4ext_rs::{conv::ClassType, types::CName};

#[derive(Debug)]
pub struct Event;

impl ClassType for Event {
    type BaseClass = IScriptable;
    const NAME: &'static str = "redEvent";
}

// instead of reimporting each of all parents methods with #[redscript_import], we probably want to upcast instead
impl Event {
    pub fn get_class_name(self: &Ref<Self>) -> CName {
        red4ext_rs::prelude::Ref::<Self>::upcast(self.clone()).get_class_name()
    }
    pub fn is_exactly_a(self: &Ref<Self>, class_name: CName) -> bool {
        red4ext_rs::prelude::Ref::<Self>::upcast(self.clone()).is_exactly_a(class_name)
    }
}
```

### CODE REVIEW

**Comment by Roms1383** [comment]
I haven't tested it yet, will do tomorrow.
That will also allow to check if I need to use `*self`, `self.deref()` or `self.clone()`.

**Comment by Roms1383** [comment]
I realize though it's convenient when inheriting for a couple of classes but far from optimal when using tons: each call to `upcast` when getting up the inheritance tree probably incurs a small performance cost on top of `#[redscript_import]`'s final call, and in that regards a bindings generator would be more adapted.

That being said, having this implementation on `IScriptable` only does not hurt much, and still provide some convenience.

What do you think, should I drop this PR ?

**Comment by Roms1383** [comment]
@jac3km4 could you give me your opinion on whether this feature is worth finishing ?

**Comment by jac3km4** [comment]
Some suggestions:
- I think these methods shouldn't take &self because that'll lead to conflict w methods on the Self type, I'd define the parameter as 'this: Self' instead
- it might make sense to just define this generically for all types: `impl <A> Ref<A>`, it's already assumed in a few places that methods are only invoked on IScriptables

File: examples/player_info/src/lib.rs [hunk 1]
```diff
@@ -23,6 +23,13 @@ fn dump_player_info(player: Ref<PlayerPuppet>) {
     info!("player vehicles: {}", player.get_unlocked_vehicles_size());
     let can_apply_breathing = PlayerPuppet::can_apply_breathing_effect(Ref::downgrade(&player));
     info!("can apply breating effect: {}", can_apply_breathing);
+    let class_name = Ref::<PlayerPuppet>::get_class_name(player.clone());
```
Comment (jac3km4) [inline examples/player_info/src/lib.rs:26]:
```suggestion
    let class_name = Ref::get_class_name(player.clone());
```

File: examples/player_info/src/lib.rs [hunk 2]
```diff
@@ -23,6 +23,13 @@ fn dump_player_info(player: Ref<PlayerPuppet>) {
...
+    info!("class name: {}", class_name);
+    let is_ent_entity = Ref::<PlayerPuppet>::is_a(player.clone(), CName::new("entEntity"));
```
Comment (jac3km4) [inline examples/player_info/src/lib.rs:28]:
```suggestion
    let is_ent_entity = Ref::is_a(player.clone(), CName::new("entEntity"));
```

File: examples/player_info/src/lib.rs [hunk 3]
```diff
@@ -23,6 +23,13 @@ fn dump_player_info(player: Ref<PlayerPuppet>) {
...
+    info!("is an entEntity: {}", is_ent_entity);
+    let is_exactly_a_player_puppet =
+        Ref::<PlayerPuppet>::is_exactly_a(player.clone(), CName::new(PlayerPuppet::NAME));
```
Comment (jac3km4) [inline examples/player_info/src/lib.rs:31]:
```suggestion
        Ref::is_exactly_a(player.clone(), CName::new(PlayerPuppet::NAME));
```

File: examples/player_info/src/lib.rs (hunk 1 above)
Comment (Roms1383) [reply examples/player_info/src/lib.rs]: Done in 36efc8c

File: examples/player_info/src/lib.rs (hunk 2 above)
Comment (Roms1383) [reply examples/player_info/src/lib.rs]: Done in 36efc8c

File: examples/player_info/src/lib.rs (hunk 3 above)
Comment (Roms1383) [reply examples/player_info/src/lib.rs]: Done in 36efc8c

**Pull request merged**